File: src/main/java/no/bekk/java/exercises/Ex2_LambdaOnCollectionsPart1.java
package no.bekk.java.exercises;

import no.bekk.java.model.League;
import no.bekk.java.model.Player;
import no.bekk.java.model.Team;

import java.time.LocalDate;
import java.util.List;

import static java.util.stream.Collectors.*;
import static no.bekk.java.util.Utils.implementThis;

public class Ex2_LambdaOnCollectionsPart1 {

	static List<String> teamNames(List<Team> teams) {
		return implementThis();
	}

	static List<Player> removeOldPlayers(LocalDate maxAge, List<Player> players) {
		return implementThis();
	}

	static List<Team> addValueToEachTeam(Double percent, List<Team> teams) {
		return implementThis();
	}

	static List<Team> removeTeamsWithLowValue(List<Team> teams, Double minValue) {
		return implementThis();
	}

	static List<League> setLeagueNamesToUpperCase(List<League> leagues) {
		return implementThis();
	}
}

File: src/main/scala/no/bekk/scala/exercises/Ex2_LambdaOnCollectionsPart1.scala
package no.bekk.scala.exercises

import java.time.LocalDate

import no.bekk.scala.model.{League, Player, Team}

object Ex2_LambdaOnCollectionsPart1 {

	def teamNames(teams: List[Team]): List[String] = teams.map(_.name)

	def removeOldPlayers(maxAge: LocalDate, players: List[Player]): List[Player] = players.filter(_.birthDate.isAfter(maxAge))

	def addValueToEachTeam(percent: Double, teams: List[Team]): List[Team] =
		teams.map(team => team.copy(value = team.value * (percent + 1)))

	def removeTeamsWithLowValue(teams: List[Team], minValue: Double): List[Team] = teams.filter(_.value >= minValue)

	def setLeagueNamesToUpperCase(leagues: List[League]): List[League] =
		leagues.map(league => league.copy(name = league.name.toUpperCase))
}
